### PR TITLE
Fix/keyp 322 transmission mode

### PR DESCRIPTION
**Pull request by jeanpierrefortune**

### CODE REVIEW

File: java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReader.java [hunk 1]
```diff
@@ -14,21 +14,4 @@
 import org.eclipse.keyple.core.seproxy.event.ObservableReader;
 
 /** The PcscReader interface provides the public elements used to manage a PC/SC reader. */
-public interface PcscReader extends ObservableReader {
-  String SETTING_KEY_TRANSMISSION_MODE = "transmission_mode";
```
Comment (pterr) [inline java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReader.java:18]: I'm not sure there is an interest to move these parameters to PcscReaderSetting

File: java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReaderSetting.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+package org.eclipse.keyple.plugin.pcsc;
+
+/**
+ * This class defines the constants useful for the configuration of the PC/SC readers.<br>
+ * These constants are strings defining the keys (prefix KEY_) and values to be used with the
+ * setParameter method.
+ */
+public final class PcscReaderSetting {
+  /** key to set the transmission mode */
+  public static final String KEY_TRANSMISSION_MODE = "transmission_mode";
```
Comment (pterr) [inline java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReaderSetting.java:21]: PcscReaderSetting is a public final class: all its public elements are static final attributes.
These parameters could still be located in the PcscReader interface.

**Review by pterr — APPROVED** [review]

File: java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReaderSetting.java (hunk 2 above)
Comment (jeanpierrefortune) [reply java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReaderSetting.java:21]: The idea is to answer this point raised by SonarQube:
https://sonarcloud.io/project/issues?id=eclipse_keyple-plugin_pcsc&open=AXMjT1DptNti-jTiWO4G&resolved=false&severities=CRITICAL&types=CODE_SMELL
The constant interface pattern is generally discouraged.

Comment (jeanpierrefortune) [reply java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReaderSetting.java:21]: @andrei-cristea , do you have an opinion or advice on this subject?

File: java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReader.java (hunk 1 above)
Comment (andrei-cristea) [reply java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReader.java:18]: Not to PcscReaderSetting, but to PcscReaderConstants

File: java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReaderImpl.java [hunk 3]
```diff
@@ -39,7 +43,7 @@
 import org.slf4j.LoggerFactory;
 
 final class PcscReaderImpl extends AbstractObservableLocalReader
-    implements PcscReader, SmartInsertionReader, SmartRemovalReader {
+    implements ObservableReader, SmartInsertionReader, SmartRemovalReader {
```
Comment (andrei-cristea) [inline java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReaderImpl.java:46]: implements PcscReader,...

**Review by andrei-cristea — CHANGES_REQUESTED** [review]

File: java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReaderSetting.java (hunk 2 above)
Comment (andrei-cristea) [reply java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReaderSetting.java]: I agree with Sonar

File: java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReaderImpl.java (hunk 3 above)
Comment (jeanpierrefortune) [reply java/component/keyple-plugin/pcsc/src/main/java/org/eclipse/keyple/plugin/pcsc/PcscReaderImpl.java]: done

**Review by andrei-cristea — APPROVED** [review]

**Pull request merged**